reporter-config3/src/main/java/com/addthis/metrics3/reporter/config/InfluxDBReporterConfig.java

```
@@ -0,0 +1,101 @@
+import com.codahale.metrics.MetricRegistry;
+import com.izettle.metrics.influxdb.InfluxDbHttpSender;
+import com.izettle.metrics.influxdb.InfluxDbReporter;
+import com.izettle.metrics.influxdb.InfluxDbSender;
+
+public class InfluxDBReporterConfig extends AbstractInfluxDBReporterConfig implements MetricsReporterConfigThree
+{
+    private static final Logger log = LoggerFactory.getLogger(InfluxDBReporterConfig.class);
+
+    private InfluxDbReporter reporter;
+
+    private void enableMetrics3(HostPort hostPort, MetricRegistry registry) throws Exception
+    {
+        InfluxDbSender influxDbSender = new InfluxDbHttpSender(getProtocol(), hostPort.getHost(), hostPort.getPort(),
+            getDbName(), getAuth(), getRealRateunit(), getConnectionTimeout(), getReadTimeout());
+
+        reporter = InfluxDbReporter.forRegistry(registry).convertRatesTo(getRealRateunit())
+            .convertDurationsTo(getRealDurationunit()).withTags(getResolvedTags())
+            .filter(MetricFilterTransformer.generateFilter(getPredicate())).build(influxDbSender);
+
+        reporter.start(getPeriod(), getRealTimeunit());
+    }
+
+    @Override
+    public void report()
+    {
+        if (reporter != null) {
+            reporter.report();
+        }
+    }
+
+    @Override
+    public boolean enable(final MetricRegistry registry)
+    {
+        boolean success = checkClass("com.izettle.metrics.influxdb.InfluxDbReporter");
+        if (!success)
+        {
+            return false;
+        }
+
+        List<HostPort> hosts = getFullHostList();
+        if (hosts == null || hosts.isEmpty())
+        {
+            log.error("No hosts specified, cannot enable InfluxDBReporter");
+            return false;
+        }
+
+        for (HostPort hostPort : hosts)
+        {
+            log.info("Enabling InfluxDBReporter to {}:{}", hostPort.getHost(), hostPort.getPort());
+            try
+            {
+                // connect with first working host
```

This doesn't appear to be how the other reporters work. It's also not very clear if I do put in multiple hosts to the config. I think it would be more clear to just require 1 host and fail hard when there are not only 1. Then you can do a HA setup behind a LB w/ influxdb-relay or something.